src/lib/core/base/base-component.ts

```diff
@@ -5,4 +5,9 @@ export interface IBaseComponent extends HTMLElement {
   attributeChangedCallback?(name: string, oldValue: string, newValue: string): void;
 }
 
-export abstract class BaseComponent extends HTMLElement implements IBaseComponent {}
+export abstract class BaseComponent extends HTMLElement implements IBaseComponent {
+  // We provide default implementation of these methods so that they can be extended by other
+  // classes.
+  public connectedCallback(): void {}
+  public attributeChangedCallback(name: string, oldValue: string | null, newValue: string | null): void {}
```

Comment:
I thought we determined that we didn't need these at one point? Am I misremembering?